src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -904,29 +942,228 @@ void BrgemmEmitter::emit_brgemm_kernel_call(const brgemm_kernel_t *brgKernel, in
+    return { in0_offset, in1_offset, in2_offset, out0_offset };
+}
+
+void BrgemmWithScratchEmitter::kernel_preparation(const brgemmCtx& ctx) const {
+    if (ctx.is_with_amx) {
+        size_t gpr_size = 8;
+        Xbyak::Operand gprs_to_save[] = {h->r8, h->r9, h->r10, h->r11, h->rax,
+                                         h->rcx, h->rdx, h->rdi, h->rsi, h->rbp, h->rbx};
+        size_t n_gprs_to_save = sizeof(gprs_to_save) / sizeof(gprs_to_save[0]);
+
+        h->sub(h->rsp, n_gprs_to_save * gpr_size);
+        for (size_t i = 0; i < n_gprs_to_save; ++i)
+            h->mov(h->ptr[h->rsp + i * gpr_size], gprs_to_save[i]);
+
+        // save function address in gpr to pass in call instruction
+        const auto& overload = static_cast<status_t(*)(const char*)>(amx_tile_configure);
+        h->mov(h->rbp, reinterpret_cast<uintptr_t>(overload));
+        h->mov(abi_param1, reinterpret_cast<uintptr_t>(ctx.palette));
+
+        // align stack on 16-byte as ABI requires
+        // note that RBX must not be changed by the callee
+        h->mov(h->rbx, h->rsp);
+        h->and_(h->rbx, 0xf);
+        h->sub(h->rsp, h->rbx);
+
+        h->call(h->rbp);
+
+        h->add(h->rsp, h->rbx);
+        // restore gpr registers
+        for (int i = n_gprs_to_save - 1; i >= 0; --i)
+            h->mov(gprs_to_save[i], h->ptr[h->rsp + i * gpr_size]);
+        h->add(h->rsp, n_gprs_to_save * gpr_size);
+    }
+}
+
+void BrgemmWithScratchEmitter::kernel_call(const brgemm_kernel_t *brg_kernel,
+                                           const std::vector<Reg64>& regs,
+                                           const std::vector<size_t>& offsets) const {
+    if (regs.size() != offsets.size() || regs.size() != 4) {
+        IE_THROW() << "Brgemm emitter got unexpected register count and offset count: " << regs.size() << " and " << offsets.size();
+    }
+
+    size_t gpr_size = 8;
+    size_t num_args_passed_on_stack = 0;
+    // save function address in gpr to pass in call instruction
+    const auto& brgemm_kernel_overload = static_cast<void (*)(const brgemm_kernel_t*,
+                                                              const void*,
+                                                              const void*,
+                                                              void*,
+                                                              void*,
+                                                              int)>(kernel_execute);
+    h->mov(h->rbp, reinterpret_cast<uintptr_t>(brgemm_kernel_overload));
+    // todo: several of addr_{A, B, C} could be also abi_paramX, so one of them could be corrupted
+    //  if moving directly h->uni_vmovq(abi_paramX, adr_X). Save them to vector regs to avoid corruption.
+    //  It's likely that a more efficient solution exists.
+    h->uni_vmovq(Xmm(0), regs[0]); // A
+    h->uni_vmovq(Xmm(1), regs[1]); // B
+    h->uni_vmovq(Xmm(2), regs[2]); // Scratch
```

Comment:
We can reorder registers in `regs` to match better with `BrgemmEmitter::kernel_call` (e.g. Scratch, if present, goes last)